src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -0,0 +1,376 @@
+            engine_args=self.engine_args,
+            engine_context=self.context,
+            sampling_temperature=self.sampling_temperature,
+            deterministic=self.deterministic,
+            sequence_length=self.sequence_length,
+            input_ids_length=prompt_processing_sequence_length,
+            tokenizer=self.tokenizer,
+            use_deepsparse_cache=use_deepsparse_cache,
+        )
+
+        if self.multitoken_engine.kv_cache_enabled:
+            # unless kv cache is enabled, we don't
+            # need to initialize the single token engine
+            self.engine = NLDecoderEngine(
+                onnx_file_path=self.onnx_file_path,
+                engine_type=self.engine_type,
+                engine_args=self.engine_args,
+                engine_context=self.context,
+                sampling_temperature=self.sampling_temperature,
+                deterministic=self.deterministic,
+                sequence_length=self.sequence_length,
+                input_ids_length=1,
+                tokenizer=self.tokenizer,
+                use_deepsparse_cache=use_deepsparse_cache,
+            )
+
+    @staticmethod
+    def route_input_to_bucket(
+        *args, input_schema: BaseModel, pipelines: List[Pipeline], **kwargs
+    ) -> Pipeline:
+        """
+        This method is used to route the input to the correct pipeline.
+
+        :param args: args to pass to the pipeline
+        :param input_schema: the input schema for the pipeline
+        :param pipelines: the list of pipelines to route the input to
+        :param kwargs: kwargs to pass to the pipeline
+        :return: the pipeline to route the input to
+        """
+        raise ValueError("Bucketing is not supported for generation pipelines")
+
+    @property
+    def input_schema(self) -> Type[BaseModel]:
+        """
+        Property to return the input schema for the pipeline.
+
+        :return: the input schema for the pipeline
+        """
+        return TextGenerationInput
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        """
+        Property to return the output schema for the pipeline.
+
+        :return: the output schema for the pipeline
+        """
+        return TextGenerationOutput
+
+    def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
+        """
+        Convert the input schema for the pipeline to the inputs for the engine.
+
+        :param inputs: the input schema for the pipeline
+        :return: the inputs for the engine
+        """
+
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+
+        input_tokens = self.tokenizer(
+            inputs.sequences,
+            return_tensors="np",
+            max_length=self.sequence_length,
+            padding="max_length",
+        )
+
+        attention_mask = input_tokens["attention_mask"]
+
+        # TODO: Positions input is not required by BLOOM
+        # let's make it optional in the future
```

Comment:
+1, we should detect if `positions` (and later `causal_mask`)  is a model input and operate accordingly

Reply:
The cool thing is that actually, L222 filters out those inputs which are not used by the graph. So we get the filtering for free esentially.